networking.md

```diff
@@ -506,4 +506,166 @@ spec:
 Additionally there is a CRD-less `service_controller`. It watches for `LoadBalancer` services in the same namespace of a `DNSMasq` instance and creates a `DNSData` CR to auto register service endpoints which have an annotation set with  `"dnsmasq.network.openstack.org/hostname": <hostname>`. This annotation gets auto added to the `LoadBalancer` services exposed using lib-common `https://github.com/openstack-k8s-operators/lib-common/pull/258`.
 
 ## IPAM
-TBD
\ No newline at end of file
+The the infra-operator privides CRDs for IPAM.
+
+The `NetConfig` CRD allows do describe the overall networks used for the environment.
+
+* a `network` described by `name`, `mtu` (default 1500), and one or multiple subnets.
+* a `subnet` describedc by `name`, one or multiple `allocationRanges` with `start` and `end`, `cidr`, `vlan` id, `gateway` and `routes`. Using `excludeAddresses` individual IP addresses from within the `cidr` can be excluded to be assigned (dynamic or static).
+
+Example `NetConfig`:
+```yaml
+apiVersion: network.openstack.org/v1beta1
+kind: NetConfig
+metadata:
+  name: netconfig
+spec:
+  networks:
+  - name: CtlPlane
+    subnets:
+    - name: subnet1
+      allocationRanges:
+      - end: 192.168.122.120
+        start: 192.168.122.100
+      - end: 192.168.122.200
+        start: 192.168.122.150
+      cidr: 192.168.122.0/24
+      gateway: 192.168.122.1
+  - name: InternalApi
+    subnets:
+    - name: subnet1
+      allocationRanges:
+      - end: 172.17.0.250
+        start: 172.17.0.10
+      excludeAddresses:
+      - 172.17.0.10
+      - 172.17.0.12
+      cidr: 172.17.0.0/24
+      vlan: 20
+  - name: External
+    subnets:
+    - name: subnet1
+      allocationRanges:
+      - end: 10.0.0.250
+        start: 10.0.0.10
+      cidr: 10.0.0.0/24
+      gateway: 10.0.0.1
+      routes:
+      - destination: 10.0.1.0/24
+        nexthop: 10.0.0.100
+  - name: Storage
+    subnets:
+    - name: subnet1
+      allocationRanges:
+      - end: 172.18.0.250
+        start: 172.18.0.10
+      cidr: 172.18.0.0/24
+      vlan: 30
+  - name: StorageMgmt
+    subnets:
+    - name: subnet1
+      allocationRanges:
+      - end: 172.19.0.250
+        start: 172.19.0.10
+      cidr: 172.19.0.0/24
+      vlan: 40
+  - name: Tenant
+    mtu: 9000
+    subnets:
+    - name: subnet1
+      allocationRanges:
+      - end: 172.20.0.250
+        start: 172.20.0.10
+      cidr: 172.20.0.0/24
+      vlan: 50
+```
+
+The `IPSet` CRD can be used by a requestor to register an IP address.
```

Comment:
Is "an IP address" correct? - It is registering IP addresses on one or more networks?

Reply:
right